apps/projects/routes.py

```diff
@@ -752,22 +674,218 @@ def delete(self, project_id):
+                      duration:
+                        type: float
+                        example: 300.014000
+                      bit_rate:
+                        type: int
+                        example: 287654
+                      nb_frames:
+                        type: int
+                        example: 7654
+                      r_frame_rate:
+                        type: string
+                        example: 24/1
+                      format_name:
+                        type: string
+                        example: mov,mp4,m4a,3gp,3g2,mj2
+                      size:
+                        type: int
+                        example: 14567890
+                  url:
+                    type: string
+                    example: http://localhost:5050/projects/5cbd5acfe24f6045607e51aa/raw/video
+                  mime_type:
+                    type: string
+                    example: video/mp4
+                  create_time:
+                    type: string
+                    example: 2019-07-02T15:02:32+00:00
+                  original_filename:
+                    type: string
+                    example: video.mp4
+                  request_address:
+                    type: string
+                    example: 127.0.0.1
+                  version:
+                    type: integer
+                    example: 1
+                  parent:
+                    type: string
+                    example: 5d2c8078fe985e7587b50de7
+                  processing:
+                    type: object
+                    properties:
+                      video:
+                        type: boolean
+                        example: False
+                      thumbnail_preview:
+                        type: boolean
+                        example: False
+                      thumbnails_timeline:
+                        type: boolean
+                        example: False
+                  thumbnails:
+                    type: object
+                    properties:
+                      timeline:
+                        type: array
+                        example: []
+                      preview:
+                        type: object
+                        example: {}
+        """
+
+        if any(self._project['processing'].values()):
+            return json_response({"processing": True}, status=409)
+
+        # deepcopy & save a child_project
+        child_project = copy.deepcopy(self._project)
+        del child_project['_id']
+        del child_project['storage_id']
+        child_project['parent'] = self._project['_id']
+        child_project['create_time'] = datetime.utcnow()
+        child_project['version'] += 1
+        child_project['thumbnails'] = {
+            'timeline': [],
+            'preview': None
+        }
+        app.mongo.db.projects.insert_one(child_project)
+
+        # put a video file stream into storage
+        try:
+            storage_id = app.fs.put(
```

Comment:
wondering why copy it on duplicate before any changes are done?

Reply:
agreed to optimised it on later stages